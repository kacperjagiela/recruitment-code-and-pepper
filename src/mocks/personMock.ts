import { Person } from '~/types';

const personMock = {
    name: 'C-3PO',
    height: 167,
    mass: 75,
    hair_color: 'n/a',
    skin_color: 'gold',
    eye_color: 'yellow',
    birth_year: '112BBY',
    gender: 'n/a',
} as Person;

export default personMock;
