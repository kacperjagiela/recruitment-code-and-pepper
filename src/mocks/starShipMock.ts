import { StarShip } from '~/types';

const starShipMock = {
    name: 'CR90 corvette',
    model: 'CR90 corvette',
    manufacturer: 'Corellian Engineering Corporation',
    cost_in_credits: 3500000,
    length: 150,
    max_atmosphering_speed: 950,
    crew: 30,
    passengers: 600,
    cargo_capacity: 3000000,
    consumables: { days: 0, months: 0, years: 1, fullString: '1 year' },
    hyperdrive_rating: 2.0,
    MGLT: 60,
} as StarShip;

export default starShipMock;
