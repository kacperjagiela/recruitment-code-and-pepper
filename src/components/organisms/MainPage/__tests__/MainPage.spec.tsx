import { render } from '@testing-library/react';

import personMock from '~/mocks/personMock';
import starShipMock from '~/mocks/starShipMock';

import { MainPage } from '..';

describe('<MainPage />', () => {
    it('renders correctly', () => {
        const { container } = render(<MainPage people={[personMock]} starShips={[starShipMock]} />);

        expect(container).toMatchSnapshot();
    });
});
