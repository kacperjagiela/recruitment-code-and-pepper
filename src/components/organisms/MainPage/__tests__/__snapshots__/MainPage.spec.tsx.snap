// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainPage /> renders correctly 1`] = `
<div>
  <div
    class="MuiBox-root css-1pe7sww"
  >
    <div
      class="MuiBox-root css-0"
    >
      <h3
        class="MuiTypography-root MuiTypography-h3 css-1imu44y-MuiTypography-root"
      >
        Battle it out! 
        <br />
        (Star Wars edition)
      </h3>
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth css-cq5a0p-MuiFormControl-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="false"
          id="select_input_Battle_type"
        >
          Battle type
        </label>
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiSelect-root css-tj4fkw-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <div
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-labelledby="select_input_Battle_type select_Battle type"
            class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
            id="select_Battle type"
            role="button"
            tabindex="0"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </div>
          <input
            aria-hidden="true"
            class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
            tabindex="-1"
            value=""
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-dhh0px"
            >
              <span>
                Battle type
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <div
        class="MuiBox-root css-1lfji6r"
      >
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root Mui-disabled css-1ft3dba-MuiButtonBase-root-MuiButton-root"
          disabled=""
          tabindex="-1"
          type="button"
        >
          Random fight!
        </button>
      </div>
    </div>
    <div
      class="MuiBox-root css-15z5ch3"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-1uwgr7b-MuiTypography-root"
      >
        Developed by
         
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1kgfegd-MuiTypography-root-MuiLink-root"
          href="https://github.com/kacperjagiela"
        >
          Kacper Jagieła
        </a>
         
        for
         
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1kgfegd-MuiTypography-root-MuiLink-root"
          href="https://codeandpepper.com/"
        >
          Code & Pepper
        </a>
      </p>
    </div>
  </div>
</div>
`;
